inbm/dispatcher-agent/tests/ai/snapshot_genai_unit_test.py

```diff
@@ -0,0 +1,1219 @@
+        with patch('dispatcher.sota.snapshot.PseudoShellRunner') as mock:
+            mock.return_value.run.return_value = ('output', 'error', 0)
+            yield mock
+
+    @pytest.fixture
+    def mock_mender_commit_command(self):
+        with patch('dispatcher.sota.snapshot.mender_commit_command') as mock:
+            mock.return_value = 'mender -commit'
+            yield mock
+
+    @pytest.mark.parametrize('cmd_output, expected', [
+        (('output', 'error', 0), 0),
+        (('output', 'error', 1), 1),
+    ])
+    def test_commit(self, mock_dispatcher_state, mock_pseudo_shell_runner, mock_mender_commit_command, cmd_output, expected):
+        # Arrange
+        mock_pseudo_shell_runner.return_value.run.return_value = cmd_output
+        yocto_snapshot = YoctoSnapshot(MagicMock(), 'sota_cmd', MagicMock(), 'snap_num', False, False)
+
+        # Act
+        result = yocto_snapshot.commit()
+
+        # Assert
+        assert result == expected
+        mock_dispatcher_state.clear_dispatcher_state.assert_called_once()
+        mock_pseudo_shell_runner.return_value.run.assert_called_once_with('mender -commit')
+
+#DO NOT DELETE THIS LINE - TestYoctoSnapshotRecover
+'''
+ADD HUMAN FEEDBACK BELOW:
+
+'''
+class TestYoctoSnapshotRecover:
+    @pytest.mark.parametrize('snap_num, reboot_device, time_to_wait_before_reboot', [
+        ('123', True, 10),  # Test with valid snap_num and reboot_device is True
+        ('123', False, 10),  # Test with valid snap_num and reboot_device is False
+        (None, True, 10),  # Test with snap_num is None and reboot_device is True
+        (None, False, 10),  # Test with snap_num is None and reboot_device is False
+        ('0', True, 10),  # Test with snap_num is '0' and reboot_device is True
+        ('0', False, 10),  # Test with snap_num is '0' and reboot_device is False
+    ])
+    @patch('dispatcher.sota.snapshot.dispatcher_state.clear_dispatcher_state')
+    @patch('time.sleep')
+    def test_recover(self, mock_sleep, mock_clear_dispatcher_state, snap_num, reboot_device, time_to_wait_before_reboot):
+        # Create a mock rebooter object
+        mock_rebooter = Mock()
+        # Create a YoctoSnapshot instance with the test parameters
+        yocto_snapshot = YoctoSnapshot(Mock(), 'update', Mock(), snap_num, False, reboot_device)
+        # Call the recover method with the test parameters
+        yocto_snapshot.recover(mock_rebooter, time_to_wait_before_reboot)
+        # Verify that the clear_dispatcher_state function was called
+        mock_clear_dispatcher_state.assert_called_once()
+        # Verify that the sleep function was called with the correct parameter
+        mock_sleep.assert_called_once_with(time_to_wait_before_reboot)
+        # Verify that the reboot method was called on the rebooter object
+        mock_rebooter.reboot.assert_called_once()
+
+#DO NOT DELETE THIS LINE - TestYoctoSnapshotRevert
+'''
+ADD HUMAN FEEDBACK BELOW:
+
+'''
+class TestYoctoSnapshotRevert:
```

Comment:
To go back to Pull Request Summary table, Click on the [link](https://github.com/intel/intel-inb-manageability/pull/489#)

<h3>Class: YoctoSnapshot |Function: revert</h3>

[Source Code Link](https://github.com/intel/intel-inb-manageability/blob/NEX_MAXIMUS_UT/test-specific-unit-test/inbm/dispatcher-agent/dispatcher/sota/snapshot.py?plain=1#L403)

[Unit Test Case Code Link](https://github.com/intel/intel-inb-manageability/blob/NEX_MAXIMUS_UT/test-specific-unit-test/inbm/dispatcher-agent/tests/ai/snapshot_genai_unit_test.py?plain=1#L829)

<h4>Test Case Scenarios:</h4>
The unit test suite `TestYoctoSnapshotRevert` contains a single test case `test_revert`. However, thanks to the `pytest.mark.parametrize` decorator, this single test case is run multiple times with different parameters, effectively creating multiple test scenarios. Here are the test cases generated by this unit test:

1. `test_revert[0]`: This test case tests the `revert` method of the `YoctoSnapshot` class with `time_to_wait_before_reboot` set to 0. The test verifies that the dispatcher state is cleared, the system waits for 0 seconds before rebooting, and the system reboots.

2. `test_revert[5]`: This test case tests the `revert` method of the `YoctoSnapshot` class with `time_to_wait_before_reboot` set to 5. The test verifies that the dispatcher state is cleared, the system waits for 5 seconds before rebooting, and the system reboots.

3. `test_revert[10]`: This test case tests the `revert` method of the `YoctoSnapshot` class with `time_to_wait_before_reboot` set to 10. The test verifies that the dispatcher state is cleared, the system waits for 10 seconds before rebooting, and the system reboots.

In each of these test cases, the `revert` method is expected to clear the dispatcher state, wait for the specified amount of time, and then reboot the system. The test verifies these expectations by checking that the appropriate methods (`clear_dispatcher_state`, `sleep`, and `reboot`) are called with the correct arguments. If any of these methods are not called as expected, the test will fail.

<h4>Short summary about the testcases:</h4>
No failed cases

<h4>Test Execution Results:</h4>

| Total Cases    | 3  |
| -------------- | -- |
| Passed Cases   | 3  |
| Failed Cases   | 0  |



To go back to Pull Request Summary table, Click on the [link](https://github.com/intel/intel-inb-manageability/pull/489#)